### ISSUE TITLE
Can't debug single line lambda in IntelliJ

### ISSUE THREAD

**Issue by ScottPierce** [flutter/flutter-intellij#458]
I'm having some difficulty debugging single line lambdas in IntelliJ. This doesn't seem to be a supported feature at the moment.

## Example
```dart
return document.children
          .where((node) => node.text == "channel")
          ...
```

If I set a breakpoint on the second line in the above sample, the breakpoint is triggered on the `.where(...` call, but not the lambda.

## What do I expect to happen?

I'd expect that when a breakpoint is selected for a line that is ambiguous as to what code the breakpoint is meant to select, that I'd be given a menu option to choose what code that breakpoint is selecting, similar to other lambda languages that IntelliJ supports.

[Here is a GIF with an example shown at the beginning of such an experience in scala](https://d3nmt5vlzunoa1.cloudfront.net/scala/files/2015/09/ScalaLambdas1.gif)

**Comment by jacob314** [comment]
I expect we should now be able to fix this without a dart plugin change.

**Comment by feinstein** [comment]
I opened this 3 years ago: https://youtrack.jetbrains.com/issue/IDEA-328407/Add-breakpoint-selection-for-nested-lambdas-just-like-Java-has

The UI has changed ever since: https://www.jetbrains.com/help/idea/using-breakpoints.html#set-line-breakpoint

<img width="692" alt="image" src="https://github.com/user-attachments/assets/b598b66d-dbd6-4dc1-935b-152903affe01">

It would be great if we could prioritize this

**Comment by AlexV525** [comment]
@alexander-doroshko could you help evaluate whether the Dart plugin should support this, or the Flutter plugin can do this?

**Comment by alexander-doroshko** [comment]
@AlexV525 The Dart plugin includes a debugger for pure Dart applications. The Flutter plugin includes a debugger for Flutter applications. The latter is essentially a copy of the former with some minor changes, so the Flutter debugging experience is not controlled by the Dart plugin.

**Comment by AlexV525** [comment]
> [@AlexV525](https://github.com/AlexV525) The Dart plugin includes a debugger for pure Dart applications. The Flutter plugin includes a debugger for Flutter applications. The latter is essentially a copy of the former with some minor changes, so the Flutter debugging experience is not controlled by the Dart plugin.

@alexander-doroshko Yep. So regarding the issue case, it seems this should be implemented in the Dart plugin, correct?

**Comment by alexander-doroshko** [comment]
> this should be implemented in the Dart plugin, correct?

for pure Dart apps debugging (like Dart command-line apps) - in the Dart plugin

for Flutter apps debugging - in the Flutter plugin

The Flutter debugger implementation in the Flutter plugin is a copy/pasted debugger from the Dart plugin, but it also means they technically have no shared code to reuse.

That said, I’ve never investigated the technical feasibility of this feature, so I don’t know whether it’s possible. I mean I don’t know whether the Dart VM or the Flutter VM supports it in the first place.

**Comment by AlexV525** [comment]
> for pure Dart apps debugging (like Dart command-line apps) - in the Dart plugin
> 
> for Flutter apps debugging - in the Flutter plugin

Thanks for the explanation. 👍

**Comment by feinstein** [comment]
Hi @alexander-doroshko, how could we help to prioritize this so you could investigate the feasibility for this feature?

**Comment by alexander-doroshko** [comment]
@jwren @pq Do you know if it's technically possible and could be done on the plugin end without changes in the Dart VM?

**Comment by AlexV525** [comment]
@alexander-doroshko When I was trying to enable this in the Flutter plugin I got this:

<img width="932" height="634" alt="Image" src="https://github.com/user-attachments/assets/cb8d1ce6-1279-43f4-be59-3db283be5aa1" />
